app/services/current-session.js

```diff
@@ -1,171 +1,71 @@
-      this.setProperties({
-        accountContent: account,
-        userContent: user,
-        rolesContent: roles,
-        groupContent: group,
-      });
-
-      if (group && group.get('name')) {
-        this.set('userRole', group.get('name'));
-        this.set('userRoleId', group.get('id'));
-      } else {
-        this.set('userRole', 'no-access');
-        this.set('userRoleId', null);
+        this.group = await this.store.find('account-group', groupId);
       }
 
-      this.set('isPublic', this.checkPublicRights());
-      this.set('isViewer', this.checkViewRights());
-      this.set('isEditor', this.checkEditRights());
-      this.set('isAdmin', this.checkAdminRights());
-      this.set('isOverheid', this.checkGovernmentRights());
-      this.set('isOvrb', this.checkOvrbRights());
+      this.roles = get(this.session, 'data.authenticated.data.attributes.roles');
     }
-  },
-
-  checkOvrbRights() {
-    const {
-      userRoleId,
-    } = this;
-    const {
-      ovrbId,
-      adminId,
-    } = CONFIG;
-    const roles = [ovrbId, adminId];
-    return roles.includes(userRoleId);
-  },
-
-  checkGovernmentRights() {
-    const {
-      userRoleId,
-    } = this;
-    const {
-      overheidId,
-    } = CONFIG;
-    const roles = [overheidId];
-    return roles.includes(userRoleId);
-  },
-
-  checkPublicRights() {
-    const {
-      userRoleId,
-    } = this;
-    const {
-      adminId, kanselarijId, overheidId, ministerId, usersId, kabinetId,
-    } = CONFIG;
-    const roles = [adminId, kanselarijId, overheidId, ministerId, usersId, kabinetId];
-    return roles.includes(userRoleId);
-  },
+  }
 
-  checkViewRights() {
-    const {
-      userRoleId,
-    } = this;
-    const {
-      adminId, kanselarijId, overheidId, ministerId, kabinetId,
-    } = CONFIG;
-    const roles = [adminId, kanselarijId, overheidId, ministerId, kabinetId];
-    return roles.includes(userRoleId);
-  },
+  get groupUri() {
+    return this.group && this.group.uri;
+  }
 
-  checkEditRights() {
-    const {
-      userRoleId,
-    } = this;
-    const {
-      adminId, kanselarijId,
-    } = CONFIG;
-    const roles = [adminId, kanselarijId];
-    return roles.includes(userRoleId);
-  },
+  get hasValidGroup() {
+    return this.groupUri && this.groupUri !== USER;
+  }
 
-  checkAdminRights() {
-    const {
-      userRoleId,
-    } = this;
-    const {
-      adminId,
-    } = CONFIG;
-    const roles = [adminId];
-    return roles.includes(userRoleId);
-  },
+  get isAdmin() {
+    return [ADMIN].includes(this.groupUri);
+  }
 
-  checkIsDeveloper() {
-    // The keys in this whitelist are only indicative
-    // They are not used, but use them to describe who the id belongs to
-    // Get these from https://kaleidos.vlaanderen.be, and then check the user id via the ember data tab
-    const whitelist = {
-      frederik: CONFIG.frederik,
-      ben: CONFIG.ben,
-      rafael: CONFIG.rafael,
-    };
+  get isOvrb() {
```

Comment:
Future work, but I find these to be misleading. role-like naming would be better here imo, eg `canEditAgenda`, `canViewPublications`, `canEditPublications`, `canEditNewsItems`, ....

Reply:
Agree, but future work indeed.